sympy/matrices/common.py

```
@@ -2621,7 +2653,11 @@ def _matrixify(mat):
         return mat
     if hasattr(mat, 'shape'):
         if len(mat.shape) == 2:
-            return _MatrixWrapper(mat)
+            try: # test indexing behavior of matrix-like object
+                e = mat[0][0]
+                return _MatrixWrapperRowIndexing(mat)
+            except:
+                return _MatrixWrapper(mat)
```

Bare except should not be used